### PR TITLE
fix(kernel_crawler): support debian rpi.

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind bug

**Any specific area of the project related to this PR?**

/area crawler

**What this PR does / why we need it**:

Add support for debian rpi kernels.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by FedeDP** [comment]
We are finding no headers for `rpi` even if they are there: https://packages.debian.org/bookworm-backports/kernel/

File: kernel_crawler/deb.py [hunk 1]
```diff
@@ -74,8 +74,8 @@ def scan_packages(cls, stream):
                 if key in ('Provides', 'Depends'):
                     value = value.split(', ')
             except ValueError:
-                print(line)
-                raise
+                # Just skip the line if it is malformed
+                continue
```
Comment (FedeDP) [inline kernel_crawler/deb.py:78]: Malformed lines can happen; in my tests, a line `Description:` was breaking this. Just skip it.

File: kernel_crawler/debian.py [hunk 2]
```diff
@@ -31,6 +31,7 @@ def __init__(self, arch):
         mirrors = [
             deb.DebMirror('http://mirrors.edge.kernel.org/debian/', arch, repo_filter),
             deb.DebMirror('http://security.debian.org/', arch, repo_filter),
+            deb.DebMirror('http://archive.raspberrypi.com/debian/', arch, repo_filter),
```
Comment (FedeDP) [inline kernel_crawler/debian.py:34]: Debian mirror for raspberrypi.

File: kernel_crawler/debian.py [hunk 3]
```diff
@@ -58,7 +59,8 @@ def to_driverkit_config(self, release, deps):
         headers = []
         headers_rt = []
         headers_cloud = []
-        # Magic to obtain `rt`, `cloud` and normal headers:
+        headers_rpi = []
```
Comment (FedeDP) [inline kernel_crawler/debian.py:62]: Logic to handle headers for rpi.

File: kernel_crawler/debian.py [hunk 4]
```diff
@@ -83,13 +87,23 @@ def to_driverkit_config(self, release, deps):
...
             if dep.find("kbuild") != -1:
                 headers.append(dep)
                 headers_rt.append(dep)
                 headers_cloud.append(dep)
+                headers_rpi.append(dep)
 
-        return [repo.DriverKitConfig(release + "-" + self.arch, "debian", headers),
-                      repo.DriverKitConfig(release + "-rt-" + self.arch, "debian", headers_rt),
-                      repo.DriverKitConfig(release + "-cloud-" + self.arch, "debian", headers_cloud)]
+        final = []
```
Comment (FedeDP) [inline kernel_crawler/debian.py:100]: We only add headers for which we found at least 3 packages.

File: kernel_crawler/utils/download.py [hunk 5]
```diff
@@ -34,9 +34,13 @@ def get_url(url):
...
 def get_first_of(urls):
     last_exc = Exception('Empty url list')
-    for url in urls:
+    for idx, url in enumerate(urls):
         try:
-            return get_url(url)
+            content = get_url(url)
+            # If content is None and we got elements after this one,
+            # try the next ones.
+            if content is not None or idx == len(urls) - 1:
```
Comment (FedeDP) [inline kernel_crawler/utils/download.py:42]: If `get_url` returns None content, try the next one instead of returning None.

**Comment by FedeDP** [comment]
Crawled rpi kernels:
```
Building dependency tree  [####################################]  100%
{
  "debian": [
    {
      "kernelversion": "1",
      "kernelrelease": "1:6.1.21-1+rpt1-rpi-arm64",
      "target": "debian",
      "headers": [
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-kbuild-6.1_6.1.63-1+rpt1_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi1-rpi-v8_6.1.21-1+rpt1_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi1-common-rpi_6.1.21-1+rpt1_all.deb"
      ]
    },
    {
      "kernelversion": "1",
      "kernelrelease": "1:6.1.47-1+rpt2-rpi-arm64",
      "target": "debian",
      "headers": [
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi2-common-rpi_6.1.47-1+rpt2_all.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi2-rpi-v8_6.1.47-1+rpt2_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-kbuild-6.1_6.1.63-1+rpt1_arm64.deb"
      ]
    },
    {
      "kernelversion": "1",
      "kernelrelease": "1:6.1.47-1+rpt4-rpi-arm64",
      "target": "debian",
      "headers": [
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi3-common-rpi_6.1.47-1+rpt4_all.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi3-rpi-2712_6.1.47-1+rpt4_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi3-rpi-v8_6.1.47-1+rpt4_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-kbuild-6.1_6.1.63-1+rpt1_arm64.deb"
      ]
    },
    {
      "kernelversion": "1",
      "kernelrelease": "1:6.1.54-1+rpt2-rpi-arm64",
      "target": "debian",
      "headers": [
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi4-rpi-v8_6.1.54-1+rpt2_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi4-rpi-2712_6.1.54-1+rpt2_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-kbuild-6.1_6.1.63-1+rpt1_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi4-common-rpi_6.1.54-1+rpt2_all.deb"
      ]
    },
    {
      "kernelversion": "1",
      "kernelrelease": "1:6.1.58-1+rpt2-rpi-arm64",
      "target": "debian",
      "headers": [
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi6-rpi-2712_6.1.58-1+rpt2_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi6-rpi-v8_6.1.58-1+rpt2_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-kbuild-6.1_6.1.63-1+rpt1_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi6-common-rpi_6.1.58-1+rpt2_all.deb"
      ]
    },
    {
      "kernelversion": "1",
      "kernelrelease": "1:6.1.63-1+rpt1-rpi-arm64",
      "target": "debian",
      "headers": [
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi7-rpi-v8_6.1.63-1+rpt1_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-kbuild-6.1_6.1.63-1+rpt1_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi7-rpi-2712_6.1.63-1+rpt1_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi7-common-rpi_6.1.63-1+rpt1_all.deb"
      ]
    }
  ]
}
```

**Review by leogr — APPROVED** [review]

**Pull request merged**